### ISSUE TITLE
crash when generate hvx code.

### ISSUE THREAD

**Issue by cf621** [halide/Halide#3550]
Hi,

I wrote a generator for pipelines. It can build code for all platforms (including HVX offload) except HVX standalone () (target = hexagon-32-qurt-hvx_128), and got the following error

```
Assertion failed: ((i >= FTy->getNumParams() || FTy->getParamType(i) == Args[i]->getType()) && "Calling a function with a bad signature!"), function init, file ~/softwares/llvm/llvm-repo/lib/IR/Instructions.cpp, line 299.
```

If anyone knows this issue. Could you help answer it? Many thanks in advance.

**Comment by dsharletg** [comment]
Sorry for the delay, we missed this one. If this is still an issue, you'll need to provide more information. A small pipeline that reproduces the issue would be ideal.

@pranavb-ca

**Comment by andrewjong** [comment]
Hi @dsharletg , I'm getting this issue too. Can I email you about this with my code? Using Qualcomm Halide 2.1, Hexagon SDK 3.3.3.

**Comment by dsharletg** [comment]
Especially for Qualcomm Halide, I suggest you contact Qualcomm about this @dpalermo @pranavb-ca.

**Comment by dpalermo** [comment]
In this case you are getting this error due to the fact that you are trying to vectorize a floating point type.  This is not supported on Hexagon as the only vector operations are integer based.

**Comment by steven-johnson** [comment]
Could we upgrade the error message in this case to be more helpful?

**Comment by dpalermo** [comment]
Yes.  I agree. I will look into this further and see if we can improve the reporting & catch this situation earlier in the Halide side rather than let it get into generated code for LLVM to error out on.  More detail on this case in https://github.com/halide/Halide/issues/4681

**Comment by abadams** [comment]
> [!NOTE]
> **Automated triage:** this issue was flagged by Claude as likely safe to close, for the
> reason below. The reasoning was checked against the current source, and a second pass
> tried and failed to refute it, but no human verified this specific text.
> **If the reason is wrong, please reopen the issue and say so** — some fraction of these
> are expected to be wrong.

**Reason (CLOSE_DUPLICATE):** The thread reached a conclusion: the LLVM 'Calling a function with a bad signature!' assertion here comes from vectorizing a floating-point type, which HVX does not support, and the follow-up work (detect this in Halide and emit a useful error instead of letting LLVM assert) was explicitly moved to #4681 by the Qualcomm maintainer. No repro was ever provided for this issue.

**Evidence:** dpalermo's closing comment (2020-03-06): 'More detail on this case in https://github.com/halide/Halide/issues/4681'; #4681 is still open. No repro was posted here despite requests in 2019 and 2020.

Closing as a duplicate of #4681, where the remaining work (detecting float vectorization on HVX in Halide and reporting a useful error rather than tripping an LLVM assertion) is tracked. No reproducer was ever provided here.
